src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -56,40 +55,52 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
-                try
+                var storageException = e as StorageException;
+                if (storageException != null && storageException.RequestInformation != null && 
+                    storageException.RequestInformation.HttpStatusCode == 400 &&
+                    storageException.RequestInformation.ExtendedErrorInformation.ErrorCode == "InvalidQueryParameterValue")
                 {
-                    await queue.ExistsAsync(cancellationToken);
-                    _primaryCredentials = account.Credentials;
+                    // WARNING: HACK INCOMING
+                    // Premium storage accounts do not support the GetServicePropertiesAsync call, and respond with a 400 'InvalidQueryParameterValue'.
+                    // If we see this error response classify the account as a premium account
+                    account.Type = StorageAccountTypes.Premium;
+                    return;
                 }
-                catch (OperationCanceledException)
+                else
                 {
-                    throw;
+                    // If not a recognized error, the credentials are invalid
+                    string message = String.Format(CultureInfo.CurrentCulture,
+                        "Invalid storage account '{0}'. Please make sure your credentials are correct.",
+                        account.Credentials.AccountName);
+                    throw new InvalidOperationException(message);
                 }
-                catch (StorageException exception)
+            }
+
+            // test for queue support
+            IStorageQueueClient queueClient = account.CreateQueueClient();
+            IStorageQueue queue = queueClient.GetQueueReference("name");
+            try
+            {
+                await queue.ExistsAsync(cancellationToken);
+            }
+            catch (OperationCanceledException)
+            {
+                throw;
+            }
+            catch (StorageException exception)
+            {
+                WebException webException = exception.GetBaseException() as WebException;
+                if (webException != null && webException.Status == WebExceptionStatus.NameResolutionFailure)
                 {
-                    WebException webException = exception.GetBaseException() as WebException;
-                    if (webException != null && webException.Status == WebExceptionStatus.NameResolutionFailure)
-                    {
-                        string message = String.Format(CultureInfo.CurrentCulture,
-                            "Invalid storage account '{0}'. Primary storage accounts must be general "
-                            + "purpose accounts and not restricted blob storage accounts.", account.Credentials.AccountName);
-                        throw new InvalidOperationException(message);
-                    }
-                    throw;
+                    // WARNING: HACK INCOMING
```

Comment:
Why do you need this HACK comment here? I'd just remove it. We're not planning on having a better way here, right?